src/main/java/com/intel/gkl/compression/IntelDeflater.java

```diff
@@ -139,9 +139,12 @@ public void setInput(byte[] b, int off, int len) throws NullPointerException {
             throw new NullPointerException("Input is null");
         }
         if(len <= 0) {
-            throw new NullPointerException("Input buffer length is zero.");
+            throw new NullPointerException("Input buffer length is less or equal zero.");
+        }
+        if(off < 0 || off > len )
+        {
+            throw new IndexOutOfBoundsException("Offset cannot be less then 0 and greater then length.");
```

Comment:
Check if IllegalArgumentException makes sense for consistency, for chekcing expected value ranges for `off` and `level`

Reply:
I think the function declaration will need a new "throws"  update to include IndexOutOfBoundsException or IndexOutOfBoundsException is added. 

Can you confirm?

Reply:
Hi, I added throws declaration, @Kmannth  should i change IndexOutOfBoundsException and  NullPointerException from line 142 to IllegalArgumentException?

Reply:
@mkuligow  as discussed in the call today lets move this type of error to IllegalArgumentException.